dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java

```diff
@@ -529,4 +531,52 @@ void createTestParameterFiles(Workflow workflow, String id, String branchName, W
      * @param version
      */
     abstract String getCommitID(String repositoryId, Version version);
+
+    /**
+     * Returns a workflow version with validation information updated
+     * @param version Version to validate
+     * @param entry Entry containing version to validate
+     * @param mainDescriptorPath Descriptor path to validate
+     * @return Workflow version with validation information
+     */
+    public WorkflowVersion versionValidation(WorkflowVersion version, Workflow entry, String mainDescriptorPath) {
+        Set<SourceFile> sourceFiles = version.getSourceFiles();
+        SourceFile.FileType identifiedType = entry.getFileType();
+        Optional<SourceFile> mainDescriptor = sourceFiles.stream().filter((sourceFile -> Objects
+                .equals(sourceFile.getPath(), mainDescriptorPath))).findFirst();
+
+        if (mainDescriptor.isPresent()) {
+            ImmutablePair validDescriptorSet = LanguageHandlerFactory.getInterface(identifiedType).validateWorkflowSet(sourceFiles, mainDescriptorPath);
+            Validation descriptorValidation = new Validation(identifiedType, validDescriptorSet);
+            version.addOrUpdateValidation(descriptorValidation);
+        } else {
+            Map<String, String> validationMessage = new HashMap<>();
+            validationMessage.put(mainDescriptorPath, "Missing the primary descriptor.");
+            ImmutablePair noPrimaryDescriptor = new ImmutablePair(false, validationMessage);
+            Validation noPrimaryDescriptorValidation = new Validation(identifiedType, noPrimaryDescriptor);
+            version.addOrUpdateValidation(noPrimaryDescriptorValidation);
+        }
+
+        SourceFile.FileType testParameterType = SourceFile.FileType.CWL_TEST_JSON;
```

Comment:
This is a workflow, what happens with Nextflow here?

Reply:
Oops, NFL doesn't have test parameter files (though there is an enum for it). I'll make sure it won't do parameter validation for NFL.